service_capacity_modeling/models/org/netflix/ddb.py

```diff
@@ -0,0 +1,584 @@
+import math
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from pydantic import BaseModel
+from pydantic import Field
+
+from service_capacity_modeling.interface import AccessConsistency, certain_float
+from service_capacity_modeling.interface import AccessPattern
+from service_capacity_modeling.interface import CapacityDesires
+from service_capacity_modeling.interface import CapacityPlan
+from service_capacity_modeling.interface import CapacityRequirement
+from service_capacity_modeling.interface import certain_int
+from service_capacity_modeling.interface import Clusters
+from service_capacity_modeling.interface import Consistency
+from service_capacity_modeling.interface import DataShape
+from service_capacity_modeling.interface import Drive
+from service_capacity_modeling.interface import FixedInterval
+from service_capacity_modeling.interface import GlobalConsistency
+from service_capacity_modeling.interface import Instance
+from service_capacity_modeling.interface import Interval
+from service_capacity_modeling.interface import QueryPattern
+from service_capacity_modeling.interface import RegionContext
+from service_capacity_modeling.interface import Requirements
+from service_capacity_modeling.interface import Service
+from service_capacity_modeling.interface import ServiceCapacity
+from service_capacity_modeling.models import CapacityModel
+
+
+class NflxDynamoDBArguments(BaseModel):
+    number_of_regions: int = Field(
+        default=1,
+        description="How many regions the dynamoDB table needs to be created. "
+        "This determines if we should use wcu's "
+        "or rwcu's and to correctly estimate the transfer costs",
+    )
+    transactional_write_percent: float = Field(
+        default=0,
+        description="Coordinated writes, if multiple items in single write "
+        "then all-or-nothing guarantee for writes. "
+        "Default is zero, assuming most writes are single item puts",
+    )
+    transactional_read_percent: float = Field(
+        default=0,
+        description="Coordinated reads, if multiple items in "
+        "single read then all-or-nothing guarantee for read. "
+        "Default is zero",
+    )
+    estimated_mean_item_size_bytes: int = Field(
+        default=0,
+        description="Estimated avg item size. If not supplies, "
+        "then estimated_mean_write_size_bytes from desires will "
+        "be used",
+    )
+
+
+class _ReadPlan(BaseModel):
```

Comment:
If you're returning these from the model as json anywhere you probably want to inherit from ExcludeUnsetModel

Reply:
These are not returned as json only used internally to pass b/w internal methods.